NewPlatform.Flexberry.ServiceBus.RabbitMQ/Components/SendingManager/RmqSendingManager.cs

```diff
@@ -267,6 +267,177 @@ public int GetCurrentMessageCount(string clientId, string messageTypeId)
             return this._managementClient.GetQueueAsync(queueName, this._vhost).Result.Messages;
         }
 
+        /// <summary>
+        /// Получить информацию о сообщениях, которые есть, но еще не отправлены указанному клиенту.
+        /// </summary>
+        /// <param name="clientId">Идентификатор клиента.</param>
+        /// <param name="maxCount">Максимальное количество возвращаемых записей. Если равно 0, возвращается информация о всех имеющихся сообщениях.</param>
+        /// <returns>Информация о сообщениях. Записи отсортированы в планируемом порядке отправки.</returns>
+        public IEnumerable<ServiceBusMessageInfo> GetMessagesInfo(string clientId, int maxCount = 0)
+        {
+            string queueNamePrefix = _namingManager.GetClientQueuePrefix(clientId);
+            IEnumerable<Queue> queues = _managementClient.GetQueuesAsync().Result;
+            IEnumerable<Queue> clientQueues = queues.Where(x => x.Name.StartsWith(queueNamePrefix));
+
+            List<ServiceBusMessageInfo> rmqMessagesInfo = new List<ServiceBusMessageInfo>();
+            foreach (Queue clientQueue in clientQueues)
+            {
+                var messages = _managementClient.GetMessagesFromQueueAsync(clientQueue, new GetMessagesCriteria(clientQueue.Messages, Ackmodes.ack_requeue_true)).Result;
+                foreach (var message in messages)
+                {
+                    ServiceBusMessageInfo msg = new ServiceBusMessageInfo
+                    {
+                        // TODO: Добавить заполнение других свойств.
+                        MessageTypeID = _namingManager.GetMessageType(message.RoutingKey),
+                    };
+
+                    rmqMessagesInfo.Add(msg);
+                }
+
+                if (maxCount > 0 && rmqMessagesInfo.Count >= maxCount)
+                {
+                    rmqMessagesInfo.Take(maxCount);

```

Comment:
The `Take` method does not change the original sequence.
```suggestion
                    rmqMessagesInfo = rmqMessagesInfo.Take(maxCount);
```